text/0693-property-injection.md

```diff
@@ -0,0 +1,468 @@
+
+Let's say our org wants to prevent publically accessible S3 Buckets.  You can extend the Bucket class, set `blockPublicAccess: BlockPublicAccess.BLOCK_ALL` and tell all development teams to use our new class.  Or we can also require all development teams to use `blockPublicAccess: BlockPublicAccess.BLOCK_ALL` in their code.  With Blueprint, we no longer have to subclass Bucket and it is easy for development teams to use.
+
+What are the key pieces of Blueprint?
+
+`IPropertyInjector` - An IPropertyInjector defines a way to inject additional properties that are not specified in the props.  It is specific to one Construct and operates on that Construct’s properties.  This will be called `Injector` for short.
+
+`propertyInjectors` - A collection of Injectors attached to App, Stage, or Stack.  Any matching Construct created in this App, Stage, or Stack will have the Injector applied to it.
+
+What is the Blueprint design philosophy?
+
+An Org sets the standards and default value, and it is the responsibility of the development teams to adhere to that.  We also recognize that there are situations where the development team might need to override the standard.  In these situations, we value the autonomy of the development team by giving them the flexibility to override.
+
+In the Working Backwards example below, the Org defines standard defaults for development teams to use, but development teams can override the defaults if they need to.  Development teams can also write their own Injectors and share them across teams.  
+
+## Working Backwards
+
+### Development Team Experience
+
+Development teams can start using Blueprint by attaching propertyInjectors to App.
+In the example below, this dev team wants use the Property Injectors provided by its org for S3 Buckets and Lambda Fucntions. 
+
+```ts
+import { BucketPropsInjector, FunctionPropsInjector } from '@my-org/standard-props-injectors';
+...
+const app = new App({
+  ...
+  propertyInjectors: [
+    new BucketPropsInjector(),
+    new FunctionPropsInjector(),
+  ],
+});
+```
+
+Dev teams can also attach propertyInjectors to Stacks as well.  This will override the Bucket Injector provided in App.
+
+```ts
+import { BetaBucketPropsInjector } from '@my-team/custom-props-injectors';
+...
+const betaStack = new Stack(stage, 'MyApp-beta-stack', {
+  env: { account: '123456789012', region: 'us-east-1' },
+  propertyInjectors: [
+    new BetaBucketPropsInjector(),
+  ],
+});
+```
+
+In this case, Constructs created in betaStack will use `BetaBucketPropsInjector` instead of `BucketPropsInjector`.  This is useful when Beta resources have different requirements. 
+
+See [Scope Tree Traversal](#scope-tree-traversal) for more information.
+
+### Organization Level Standardization
+
+As an org with many dev teams, we can standardize how AWS Resources are created by providing dev teams with Injectors.
+
+AWS CDK will expose this `IPropertyInjector` interface for orgs to implement:
+
+```ts
```

Comment:
Leave out this code block I think. I understand the desire to be complete and accurate, but we're purely talking about the perspective of the implementor here. Let's show a representative code snippet they would write themselves. That means we don't need to include show the library code that powers their implementation (yet).